Find a functionally equivalent code:android.media.MiniThumbFile.getMiniThumbFromFile:COMMENT
Method Modifier: public      
Comment:/**
 * Gallery app can use this method to retrieve mini-thumbnail. Full size
 * images share the same IDs with their corresponding thumbnails.
 *
 * @param id the ID of the image (same of full size image).
 * @param data the buffer to store mini-thumbnail.
 */

@@ -14,20 +14,20 @@
             byte flag = mBuffer.get();
             long magic = mBuffer.getLong();
             int length = mBuffer.getInt();
-            if (size >= 1 + 8 + 4 + length && data.length >= length) {
+            if (size >= 1 + 8 + 4 + length && length != 0 && magic != 0 && flag == 1 && data.length >= length) {
                 mBuffer.get(data, 0, length);
                 return data;
             }

